modules/tf_cloudbuild_builder/cb.tf

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  gar_uri             = "${var.gar_repo_location}-docker.pkg.dev/${var.project_id}/${local.gar_name}/${var.image_name}"
+  cloudbuild_sa       = coalesce(var.cloudbuild_sa, google_service_account.cb_sa[0].id)
+  cloudbuild_sa_email = element(split("/", local.cloudbuild_sa), length(split("/", local.cloudbuild_sa)) - 1)
+  # substitutions available in the CB trigger
+  tags_subst = {
+    "_TERRAFORM_FULL_VERSION"  = "1.1.0",
+    "_TERRAFORM_MINOR_VERSION" = "1.1",
+    "_TERRAFORM_MAJOR_VERSION" = "1",
+  }
+  img_tags_subst = [for tag in keys(local.tags_subst) : "${local.gar_uri}:v$${${tag}}"]
+
+  # extract CSR project_id and repo name for granting reader access to CSR repo
+  is_source_repo = var.dockerfile_repo_type == "CLOUD_SOURCE_REPOSITORIES"
+  # url of form https://source.developers.google.com/p/<PROJECT_ID>/r/<REPO>
+  source_repo_url_split = local.is_source_repo ? split("/", var.dockerfile_repo_uri) : []
+  source_repo_project   = local.is_source_repo ? local.source_repo_url_split[4] : ""
+  source_repo_name      = local.is_source_repo ? local.source_repo_url_split[6] : ""
+}
+
+resource "google_cloudbuild_trigger" "build_trigger" {
+  project     = var.project_id
+  name        = var.trigger_name
+  description = "Builds a Terraform runner image. Managed by Terraform."
+  source_to_build {
```

Comment:
Does this blueprint support GCS style deployment? Something that was important to NetApp